### PR TITLE
[Remove Stream/Optional usage] LPS-173754 module: item-selector/item-selector-taglib

### PR DESCRIPTION
**Pull request by JanisZhang**
Please note this module is in "Total usages". Thanks!

Ticket: https://issues.liferay.com/browse/LPS-173754

### CODE REVIEW

**Comment by JanisZhang** [comment]
ci:stop

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

File: modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/servlet/taglib/GroupSelectorTag.java [hunk 1]
```diff
@@ -128,14 +127,21 @@ private List<Group> _getGroups(HttpServletRequest httpServletRequest) {
...
 
-			_groups = groupItemSelectorProviderOptional.map(
-				groupItemSelectorProvider ->
-					groupItemSelectorProvider.getGroups(
-						group.getCompanyId(), group.getGroupId(), keywords,
-						startAndEnd[0], startAndEnd[1])
-			).orElse(
-				Collections.emptyList()
-			);
+			if (groupItemSelectorProvider == null) {
```
Comment (dantewang) [inline modules/apps/item-selector/item-selector-taglib/src/main/java/com/liferay/item/selector/taglib/servlet/taglib/GroupSelectorTag.java]: @JanisZhang Please SF this method and `_getGroupsCount` to apply "shortcut" logic.

For example, you can `return _groups` in the `if` at line 111, so you don't need to indent for the `else {` part; then you can also shortcut in this `if`, etc...

**Comment by dantewang** [comment]
@JanisZhang Please see my comment and resend, thanks~

**Pull request closed without merging**